pkg/spec/deploymentconfig.go

```diff
@@ -65,6 +65,12 @@ func (deploymentConfig *DeploymentConfigSpecMod) fixDeploymentConfig() {
 		deploymentConfig.ControllerFields.Name,
 		deploymentConfig.ControllerFields.ObjectMeta.Labels)
 
+	if deploymentConfig.ControllerFields.Appversion != "" {
+		deploymentConfig.ControllerFields.ObjectMeta.Annotations = addKeyValueToMap(appVersion,
+			deploymentConfig.ControllerFields.Appversion,
+			deploymentConfig.ControllerFields.ObjectMeta.Annotations)
+	}
```

Comment:
If you do https://github.com/kedgeproject/kedge/pull/454#discussion_r152549094 this is not needed!